x/ccv/provider/keeper/msg_server.go

```diff
@@ -309,16 +307,163 @@ func (k msgServer) SetConsumerCommissionRate(goCtx context.Context, msg *types.M
 	return &types.MsgSetConsumerCommissionRateResponse{}, nil
 }
 
-func (k msgServer) ConsumerModification(goCtx context.Context, msg *types.MsgConsumerModification) (*types.MsgConsumerModificationResponse, error) {
-	if k.GetAuthority() != msg.Authority {
-		return nil, errorsmod.Wrapf(types.ErrUnauthorized, "expected %s, got %s", k.GetAuthority(), msg.Authority)
+// CreateConsumer creates a consumer chain
+func (k msgServer) CreateConsumer(goCtx context.Context, msg *types.MsgCreateConsumer) (*types.MsgCreateConsumerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	consumerId := k.Keeper.FetchAndIncrementConsumerId(ctx)
+
+	k.Keeper.SetConsumerOwnerAddress(ctx, consumerId, msg.Signer)
+	k.Keeper.SetConsumerChainId(ctx, consumerId, msg.ChainId)
+	k.Keeper.SetConsumerPhase(ctx, consumerId, Registered)
+
+	if err := k.Keeper.SetConsumerMetadata(ctx, consumerId, msg.Metadata); err != nil {
+		return &types.MsgCreateConsumerResponse{}, errorsmod.Wrapf(types.ErrInvalidConsumerMetadata,
+			"cannot set consumer metadata: %s", err.Error())
 	}
 
+	// initialization parameters are optional and hence could be nil
+	if msg.InitializationParameters != nil {
+		if err := k.Keeper.SetConsumerInitializationParameters(ctx, consumerId, *msg.InitializationParameters); err != nil {
+			return &types.MsgCreateConsumerResponse{}, errorsmod.Wrapf(types.ErrInvalidConsumerInitializationParameters,
+				"cannot set consumer initialization parameters: %s", err.Error())
+		}
+	}
+
+	// power-shaping parameters are optional and hence could be null
+	if msg.PowerShapingParameters != nil {
```

Comment:
@insumity Update allowlist and denylist inside this block. See `UpdateConsumer`, i.e., 
```golang
    k.Keeper.UpdateAllowlist(ctx, consumerId, msg.PowerShapingParameters.Allowlist)
    k.Keeper.UpdateDenylist(ctx, consumerId, msg.PowerShapingParameters.Denylist)
```
It is fine to not update the min power for TopN (see `UpdateMinimumPowerInTopN`) as the chain is still in Registered phase. The min power will be computed and stored in state on `MakeConsumerGenesis` at spawn time. Maybe add this as a comment.